test/t8_test_cmesh_copy.c

```diff
@@ -37,33 +55,428 @@ test_cmesh_committed (t8_cmesh_t cmesh)
+      && cmesh_id < t8_get_number_of_new_empty_cmesh_testcases ()) {
+    return t8_test_create_new_empty_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_empty_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id < t8_get_number_of_new_from_class_cmesh_testcases ()) {
+    return t8_test_create_new_from_class_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_from_class_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id <
+      t8_get_number_of_new_hypercube_hybrid_cmesh_testcases ()) {
+    return t8_test_create_new_hypercube_hybrid_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_hypercube_hybrid_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id < t8_get_number_of_new_periodic_cmesh_testcases ()) {
+    return t8_test_create_new_periodic_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_periodic_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id < t8_get_number_of_new_bigmesh_cmesh_testcases ()) {
+    return t8_test_create_new_bigmesh_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_bigmesh_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id < t8_get_number_of_new_prism_cake_cmesh_testcases ()) {
+    return t8_test_create_new_prism_cake_cmesh (cmesh_id);
+  }
+  cmesh_id -= t8_get_number_of_new_prism_cake_cmesh_testcases ();
+  if (0 <= cmesh_id
+      && cmesh_id < t8_get_number_of_new_disjoint_bricks_cmesh_testcases ()) {
+    return t8_test_create_new_disjoint_bricks_cmesh (cmesh_id);
+  }
+  return t8_test_create_comm_only_cmesh (cmesh_id);
+}
+
+/* The function test_cmesh_copy (int cmesh_id,sc_MPI_Comm comm) runs the cmesh_copy test for one given cmesh,
+ * that we get through its id by caling t8_test_create_cmesh (cmesh_id). */
 static void
-test_cmesh_copy (sc_MPI_Comm comm)
+test_cmesh_copy (int cmesh_id, sc_MPI_Comm comm)
 {
-  int                 eci, retval;
+  int                 retval;
   t8_cmesh_t          cmesh_original, cmesh_copy;
 
-  for (eci = T8_ECLASS_ZERO; eci < T8_ECLASS_COUNT; ++eci) {
-    t8_global_productionf ("Testing eclass %s.\n", t8_eclass_to_string[eci]);
-
-    /* Create new hypercube cmesh */
-    cmesh_original =
-      t8_cmesh_new_hypercube ((t8_eclass_t) eci, comm, 0, 0, 0);
-    test_cmesh_committed (cmesh_original);
-    /* Set up the cmesh copy */
-    t8_cmesh_init (&cmesh_copy);
-    /* We need the original cmesh later, so we ref it */
-    t8_cmesh_ref (cmesh_original);
-    t8_cmesh_set_derive (cmesh_copy, cmesh_original);
-    /* Commit and check commit */
-    t8_cmesh_commit (cmesh_copy, comm);
-    test_cmesh_committed (cmesh_copy);
-    /* Check for equality */
-    retval = t8_cmesh_is_equal (cmesh_copy, cmesh_original);
-    SC_CHECK_ABORT (retval == 1, "Cmesh copy failed.");
-    /* Clean-up */
-    t8_cmesh_destroy (&cmesh_copy);
-    t8_cmesh_destroy (&cmesh_original);
+  /* Create new cmesh */
+  cmesh_original = t8_test_create_cmesh (cmesh_id);
+  test_cmesh_committed (cmesh_original);
```

Comment:
Did you compile your test? My compiler suggests, that you mean t8_test_cmesh_committed